amlb/utils/process.py

```diff
@@ -104,7 +116,19 @@ def communicate(process, input=None, timeout=None):
         except:  # also handles kb interrupts
             process.kill()
             raise
+
         retcode = process.poll()
+        if retcode is None:
```

Comment:
As far as I can tell, this is a reliable way to tell the process is still running. And if the process is still running at this point I think the only reason can be that the communicate function returned early, which should only happen with an activity timeout.